extensions/typescript-language-features/src/languageFeatures/quickFix.ts

```diff
@@ -239,13 +239,17 @@ class TypeScriptQuickFixProvider implements vscode.CodeActionProvider<VsCodeCode
 			return;
 		}
 
-		const fixableDiagnostics = await this.supportedCodeActionProvider.getFixableDiagnosticsForContext(context);
+		let fixableDiagnostics = await this.supportedCodeActionProvider.getFixableDiagnosticsForContext(context);
 		if (!fixableDiagnostics.size || token.isCancellationRequested) {
 			return;
 		}
 
 		if (this.client.bufferSyncSupport.hasPendingDiagnostics(document.uri)) {
-			return;
+			await this.formattingConfigurationManager.ensureConfigurationForDocument(document, token);
```

Comment:
Testing https://github.com/microsoft/vscode-pull-request-github/issues/5524
I think your PR makes sense though is it possible that more pending diagnostics occur after this if statement?

Reply:
This may work because it adds a delay but doesn't seem like the a good change. Instead maybe we could block until the diagnostics are updated?